### DISCUSSION TITLE
Forcing changeSet for property / lifecycle

### DISCUSSION THREAD

**Discussion by jasonwilson** [mikro-orm/mikro-orm#4073] [Q&A]
We have a EventListener that updates and sets a meta property on an entity when a property has changed. We need to force run this change event on each entity via a script. 

Currently setting the property to the same value and persisting doesn't cause a changeSet and the lifecycle events are not run. 

Is there away to force a property changeSet in MikroORM?

**Comment by B4nan** [comment]
Why do you want to trigger it if the value did not change?

Maybe provide some code so I can better understand what you want.

**Comment by jasonwilson** [comment]
```
@Entity()
class TestEntity {
  @PrimaryKey()
  id: string

  @EncryptedProperty({
    nullable: true,
    type: 'string',
    encrypt: {
      store: '_encryptedStore'
  })
  somePropety: string

  @Property({
    nullable: true,
    type: 'json'
  })
  _encryptedStore: EncryptedStore
}
```

We're introducing a new encrypted property into our system that uses EventListener hooks to encrypt values when they've changed. The existing values in the database are _not_ encrypted so we want to build a migration script that will utilize the existing lifecycle functionality to apply the business logic. 

I've temporarily reproduced some of the logic for the initial encryption but thought it'd be slick to plumb it entirely through mikro.

**Reply by B4nan** [reply]
You call it yourself, it's a migration script - it is termporal, so make it dumb, mark all rows first as unprocessed, and do a query updating the hash + changing this bool toggle. I'd even say using raw queries instead of EM/UoW is better here, but using UoW might help you with batching for free - that itself does not mean you need some special features of the ORM.

I don't plan to allow updates on unchanged properties - that is rather a bug I keep solving again and again, so going the other way is very improbable :]

**Reply by B4nan** [reply]
I also don't really understand why the field should not change? If you have an unencrypted field, it _will_ change to the encrypted one, or not? That's why I was asking for some code, but the entity definition you provided won't help much with that.
